Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -589,18 +586,96 @@ PlatformImagePtr ImageDecoderCG::createFrameImageAtIndex(size_t index, Subsampli
     return maskedImage ? maskedImage : image;
 }
 
+String ImageDecoderCG::decodeUTI(const SharedBuffer& data) const
+{
+    auto uti = String(CGImageSourceGetType(m_nativeDecoder.get()));
+    if (uti != "public.heif"_s && uti != "public.heic"_s)
+        return uti;
+
+    if (data.size() < sizeof(unsigned)) {
+        ASSERT_NOT_REACHED();
```

Comment:
Why is this an assert? Servers could totally just reply with 3 bytes if they want

Reply:
Oh maybe the reason is because we are confident that CGImageSourceGetType() will never return heif or heic for any file that’s less than 4 bytes long?

Okay, that makes sense.